x-pack/plugins/observability/public/pages/rule_details/index.tsx

```diff
@@ -105,8 +106,9 @@ export function RuleDetailsPage() {
   useEffect(() => {
     if (ruleTypes.length && rule) {
       const matchedRuleType = ruleTypes.find((type) => type.id === rule.ruleTypeId);
+      setRuleType(matchedRuleType);
```

Comment:
@fkanout I am wondering why matchedRuleType was being set only when `rule.consumer === ALERTS_FEATURE_ID && matchedRuleType && matchedRuleType.producer`. Does my refactoring make sense to you? 

I looked into this, because actually we could get the rule type name from `ruleType`. But it was only conditionally being set as I mentioned above, that's why I used the ruleTypeIndex to get this information.

Reply:
@mgiota, I think it should be fine if you need `matchedRuleType` before the check. 
The check is done because the `find` operator could return `undefined`, and the TS will compline if we don't check it or force it with `!`?

Reply:
Let me explain what I mean. The solution to get the rule type name is following:

```
 <ItemValueRuleSummary itemValue={ruleTypeIndex.get(rule.ruleTypeId)?.name || rule.ruleTypeId} />
```

It could be:

 `<ItemValueRuleSummary itemValue={ruleType.name || rule.ruleTypeId />`

if `ruleType` was being set properly (by calling setRuleType(matchedRuleType);). But looking in your code, you only setRuleType conditionally. It doesn't really matter, cause as I mentioned I used another way to retrieve the information I need. The question was mainly why do you conditionally set the ruleType? Wouldn't it be nice to always have ruleType available?

Reply:
Good point, @mgiota! I know it's out of the scope of this PR, but would you please proceed with your solution to avoid a separate PR for this?